### PR TITLE
fix: add loader for query builder search, also add error handling

### PR DESCRIPTION
**Pull request by dnazarenkoo**
Close: [#3255](https://github.com/SigNoz/signoz/issues/3255)

### CODE REVIEW

File: frontend/src/hooks/queryBuilder/useFetchKeysAndValues.ts [hunk 1]
```diff
@@ -151,10 +163,19 @@ export const useFetchKeysAndValues = (
 		}
 	}, [data?.payload?.attributeKeys, status]);
 
+	useEffect(() => {
```
Comment (palashgdev) [inline frontend/src/hooks/queryBuilder/useFetchKeysAndValues.ts:166]: can we make use of onSuccess here

File: frontend/src/hooks/queryBuilder/useFetchKeysAndValues.ts [hunk 2]
```diff
@@ -120,6 +125,13 @@ export const useFetchKeysAndValues = (
 			const values = Object.values(payload).find((el) => !!el) || [];
 			setResults(values);
 		}
+
+		if (error) {
+			console.error(error);
```
Comment (palashgdev) [inline frontend/src/hooks/queryBuilder/useFetchKeysAndValues.ts:130]: console

File: frontend/src/hooks/queryBuilder/useFetchKeysAndValues.ts (hunk 1 above)
Comment (dnazarenkoo) [reply frontend/src/hooks/queryBuilder/useFetchKeysAndValues.ts:166]: sure, I'll move to the useQuery for this request

Comment (dnazarenkoo) [reply frontend/src/hooks/queryBuilder/useFetchKeysAndValues.ts]: updated

**Comment by YounixM** [comment]
@dnazarenkoo: Thanks for addressing the review comments and sorry for the delay in getting this merged. Can you please resolve the conflicts and we can merge the PR post that.

**Pull request closed without merging**